### ISSUE TITLE
Runtime config stack traces do not point to a helpful location

### ISSUE THREAD

**Issue by dantswain** [elixir-lang/elixir#12385]
### Elixir and Erlang/OTP versions

Elixir 1.14.2, Erlang/OTP 24

### Operating system

Linux, macOS

### Current behavior

We are decoding some json in our `config/runtime.exs` and if the json is invalid, we get a stacktrace that does not really point to the source of the error unless you already have that context.

This is fairly easily reproducible.  Take any project with Jason as a dependency (I don't think this is specific to jason but this was the context for us) and add something like this to `config/runtime.exs`:

```
Jason.decode!("")
```

Outputs a stack trace that looks like:

```
** (Jason.DecodeError) unexpected end of input at position 0
    (jason 1.4.0) lib/jason.ex:92: Jason.decode!/2
    (stdlib 3.17.2.1) erl_eval.erl:685: :erl_eval.do_apply/6
    (stdlib 3.17.2.1) erl_eval.erl:446: :erl_eval.expr/5
    (stdlib 3.17.2.1) erl_eval.erl:123: :erl_eval.exprs/5
    (elixir 1.14.2) lib/code.ex:425: Code.validated_eval_string/3
    (elixir 1.14.2) lib/config.ex:288: Config.__eval__!/3
```

### Expected behavior

It would be very helpful to have the exact line causing the error in the stacktrace (or otherwise in the error message).  Or if that's not feasible, maybe the error message could indicate somehow that this was a problem that happened while evaluating `config/runtime.exs`?  It was relatively obvious to me, and I have fairly clear ideas how to improve this in our own code, but for a coworker who is new to elixir it was pretty opaque so I thought it would be worth raising the issue to spark discussion.

**Comment by josevalim** [comment]
This is fixed in Elixir v1.14 as long as you are running on Erlang/OTP 25 (and with an Elixir version compiled for Erlang/OTP 25).

**Comment by dantswain** [comment]
Awesome thanks!
